packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -172,7 +180,25 @@ export function buildCart(engine: CommerceEngine, props: CartProps = {}): Cart {
 
     updateItem(item: CartItem) {
       if (isNewQuantityDifferent(item.productId, item.quantity)) {
-        // TODO LENS-1497: log ec.cartAction; if new quantity > previous, 'add', otherwise, 'remove'.
+        const prevItem = itemSelector(getState(), item.productId);
+
+        const {quantity: currentQuantity, ...product} = item;
+        const action = isCurrentItemQuantityGreater(item, prevItem)
+          ? 'add'
+          : 'remove';
+        const quantity = !prevItem
+          ? currentQuantity
+          : Math.abs(currentQuantity - prevItem.quantity);
+        const currency = engine.state.commerceContext
+          .currency as CurrencyCodeISO4217;
```

Comment:
The enforcement of the type should be done in the state definition, not here.

Reply:
Fair enough! Can you add a comment `//TODO LENS-1589` there then, just to flag it up :D